docs/proposals/query-blocking.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "XXXX: Query blocking"
+description: "Query blocking"
+draft: true
+---
+
+# XXXX: Query blocking
+
+**Author:** @wilfriedroset
+
+**Date:** 05/2023
+
+**Sponsor(s):** @username of maintainer(s) willing to shepherd this MID
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:**
+
+---
+
+## Background
+
+The `query-frontend` is a component that is responsible for query acceleration through splitting, sharding, caching, and other techniques. It is also responsible for enforcing limits on queries such as total query time range.
+
+## Problem Statement
+
+In certain situations, you may not be able to control the queries being sent to your Mimir installation. These queries may be intentionally or unintentionally expensive to run, and they may affect the overall stability or cost of running your service.
+
+Example:
+
+```
+sum(increase(tfe_run_current_count{status=~"applied|planned_and_finished", instance="host1", organization_name=~".*", workspace_name=~".*"}[2674800s]) > 0)
+```
+
+The query above can be harmful as the wildcard regex would match any label values including empty string. The result of such query is not cached by Mimir. As such the read path would be under pressure if the query is repeated by the tenant (e.g: dashboard in auto repeat).
```

Comment:
> The result of such query is not cached by Mimir.

What's the reason for this? nothing strikes me as uncacheable here, maybe the large period of range selector?

Reply:
Our analysis was wrong about this point. Mimir was under too much pressure.
However, the need for this feature remains, I've updated the proposal to remove this misleading paragraph.